cpp/src/arrow/tensor.cc

```diff
@@ -51,6 +55,11 @@ static void ComputeColumnMajorStrides(const FixedWidthType& type,
     const std::vector<int64_t>& shape, std::vector<int64_t>* strides) {
   int64_t total = type.bit_width() / 8;
   for (int64_t dimsize : shape) {
+    if (dimsize == 0) {
+      return;
+    }
+  }
+  for (int64_t dimsize : shape) {
```

Comment:
I think in both of these cases the length of the strides needs to equal the length of the shape; question is what should the strides be? In NumPy they're equal to the bit width:

```
In [3]: arr = np.empty((0, 0, 0))

In [4]: arr.strides
Out[4]: (8, 8, 8)

In [5]: arr = np.empty((0,))

In [6]: arr.strides
Out[6]: (8,)
```